Fix treemap sorting

mrawdon opened the pull request:
The node size property does not exist  when sorting is performed in the current code.  From the [d3-hierarchy docs](https://github.com/d3/d3-hierarchy#node_sum) sum should be called first to set the value property.  Then sort can be called to sort the node on value.

I also used the reccomended sort functions from that doc. We may want to move sortFunction to a prop allowing a user to override these.


Adjusting the sorts, created treemaps that are more organized, flowing from largest to smallest

Now
![image](https://user-images.githubusercontent.com/1678756/30435141-0441bf6e-9937-11e7-810b-71e222f355cd.png)


vs
Before

![image](https://user-images.githubusercontent.com/1678756/30435127-fa695a60-9936-11e7-95fb-eab8a37f220e.png)

mrawdon commented:
I updated the treemap test I broke. I replaced the hard coded strings with functions to sanity check things.  I can move it back to a hard coded string if that is preferred.

mcnuttandrew commented on src/treemap/index.js:
```diff
@@ -141,8 +141,8 @@ class Treemap extends React.Component {
           .size([innerWidth, innerHeight])
           .padding(padding);
       const structuredInput = hierarchy(data)
-        .sort((a, b) => a.size - b.size)
```
it's weird that we weren't getting an error if it didn't exist?

mrawdon replied:
It would just end up with undefined-undefined=NaN which I think just leaves things in its original order

mcnuttandrew commented on tests/components/treemap-tests.js:
```diff
@@ -46,19 +46,30 @@ const TREEMAP_PROPS = {
   const $ = mount(<Treemap {...TREEMAP_PROPS}/>);
   t.equal($.find('.rv-treemap__leaf').length, 21, 'should find the right number of children');
-  const expectedText = 'EasingNeonateinterpolateISchedulableParallelPauseFunctionSequenceSequenceTransitionTransitionerTransitionEventSchedulerArrayInterpolatorColorInterpolatorDateInterpolatorInterpolatorMatrixInterpolatorNumberInterpolatorObjectInterpolatorPointInterpolatorRectangleInterpolator';
+  const expectedText = toString(TREEMAP_PROPS.data.children, 0);
```
looks like this 0 is unused?

mcnuttandrew replied:
also i'd maybe advise against having these functions generate test results? It makes it harder to update the tests?

mrawdon replied:
Yeah you're right about the 0, I'll just move those back to strings instead of the functions anyway.

mcnuttandrew commented on src/treemap/index.js:
```diff
@@ -152,8 +152,8 @@ class Treemap extends React.Component {
-      .sort((a, b) => a.size - b.size)
-      .sum(d => d.size);
+      .sum(d => d.size)
+      .sort((a, b) => b.height - a.height || b.value - a.value);
```
parens please, eg `(b.height - a.height) || (b.value - a.value)`

mrawdon commented:
I think I've addressed your concerns here.  The original test keeps the same behavior, while allowing a user to override the sortFunction which I added a test for.
